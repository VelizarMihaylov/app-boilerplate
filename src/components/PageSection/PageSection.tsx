import React from 'react'
import PropTypes from 'prop-types'

import styled from '@emotion/styled'

import { ContentWrapper } from 'components/ContentWrapper'
import { gridUnits } from 'mixin'

type ContentWrapperProps = {
  children?: React.ReactNode
}

const StyledPageSection = styled.section`
  margin-bottom: ${gridUnits(1.5)}rem;
`

export const PageSection: React.FC<ContentWrapperProps> = ({ children }) => {
  return (
    <StyledPageSection>
      <ContentWrapper>{children}</ContentWrapper>
    </StyledPageSection>
  )
}

PageSection.propTypes = {
  children: PropTypes.node
}
