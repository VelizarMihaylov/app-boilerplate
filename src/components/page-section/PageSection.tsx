import React from 'react'
import PropTypes from 'prop-types'

import styled from '@emotion/styled'

import { ContentWrapper } from 'components/content-wrapper'
import { gridUnits } from 'mixins'

type PageSectionProps = {
  children?: React.ReactNode
}

const StyledPageSection = styled.section`
  margin-bottom: ${gridUnits(1.5)}rem;
`

export const PageSection: React.FC<PageSectionProps> = ({ children }) => {
  return (
    <StyledPageSection>
      <ContentWrapper>{children}</ContentWrapper>
    </StyledPageSection>
  )
}

PageSection.propTypes = {
  children: PropTypes.node
}
