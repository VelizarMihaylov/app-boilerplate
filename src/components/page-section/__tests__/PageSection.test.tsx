import React from 'react'
import { create } from 'react-test-renderer'

import { PageSection } from '../PageSection'

describe('PageSection', () => {
  it('should render without throwing an error', () => {
    const renderer = create(<PageSection />)

    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
