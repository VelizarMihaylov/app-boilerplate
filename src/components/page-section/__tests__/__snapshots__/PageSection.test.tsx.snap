// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection should render without throwing an error 1`] = `
<section
  className="css-180vpgq"
>
  <div
    className="css-1ocmp06"
  />
</section>
`;
